openhtmltopdf-core/src/main/java/com/openhtmltopdf/swing/NaiveUserAgent.java

```diff
@@ -662,7 +664,7 @@ public String resolveURI(String baseUri, String uri) {
             // Fix for OpenHTMLtoPDF issue-#125, URI class doesn't resolve jar: scheme urls and so returns only
             // the relative part on calling base.resolve(relative) so we use the URL class instead which does
             // understand jar: scheme urls.
-            URL base = new URL(baseUri);
+            URL base = new URI(baseUri).toURL();
             URL absolute = new URL(base, uri);
```

Comment:
yeah, I guess we still have to use the URL constructor here. I looked for a while and couldn't find an alternative.

Reply:
this constructor is not deprecated so it should be fine?

Reply:
We cannot utilize APIs introduced after Java 8 because we need to compile to that target.

Reply:
These URLs point to more than just files. Would Paths be sufficient here?

Reply:
@XSpielinbox @siegelzc Please review